packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -216,6 +224,20 @@ class TimeInput extends React.Component {
     });
   }
 
+  handleMissingHourChange(prevState) {
+    let hour = TimeUtil.splitHour(this.props.value);
+    if (this.props.variant === TimeUtil.FORMAT_12_HOUR) {
+      hour = TimeUtil.parseTwelveHourTime(hour, this.anteMeridiem, this.postMeridiem).hourString;
+    }
+    const minute = TimeUtil.splitMinute(this.props.value);
```

Comment:
This could be done incrementally as well without having to parse everything at once but maybe what you have now is more readable, tho.

```
let hour = TimeUtil.splitHour(this.props.value);
if (this.props.variant === TimeUtil.FORMAT_12_HOUR) {
   hour = TimeUtil.parseTwelveHourTime(hour, this.anteMeridiem, this.postMeridiem).hourString;
}

if (hour === prevState.hour) {
   const minute = TimeUtil.splitMinute(this.props.value);
   if (minute === prevState.minute) {
      const second = TimeUtil.splitSecond(this.props.value);
      return second !== prevState.second;
    }
} 

return true;
```

Reply:
As you mentioned 😃  for readability purpose I would like to keep this condition the way it is now.